library/system/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,301 @@
+    #
+    # When the service (:on_boot) and the socket (:on_demand) are enabled, the start mode is translated
+    # to :on_boot.
+    #
+    # @return [Symbol] Start mode (:on_boot, :on_demand, :manual)
+    def start_mode
+      new_value_for(:start_mode) || current_start_mode
+    end
+
+    # Sets the service start mode
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand). The given value will be applied after calling #save.
+    #
+    # @see #start_modes
+    # @raise ArgumentError when mode is not valid
+    def start_mode=(mode)
+      if !start_modes.include?(mode)
+        raise ArgumentError, "Invalid start mode: '#{mode}' for service '#{service.name}'"
+      end
+
+      if mode == current_start_mode
+        unregister_change(:start_mode)
+      else
+        register_change(:start_mode, mode)
+      end
+    end
+
+    # Determine whether the service will be active after calling #save
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
+    def active?
+      return new_value_for(:active) if changed_value?(:active)
+      service.active?
+    end
+
+    # Sets the service to be started after calling #save
+    #
+    # @see #active=
+    def start
+      self.active = true
+    end
+
+    # Sets the service to be stopped after calling #save
+    #
+    # @see #active=
+    def stop
+      self.active = false
+    end
+
+    # Toggles the service status
+    #
+    # @see #active=
+    def toggle
+      self.active = !active?
+    end
+
+    # Saves changes to the underlying system
+    #
+    # @param set_status [Boolean] Do not change service status. Useful when running on 1st stage.
+    def save(ignore_status: false)
+      save_start_mode
+      set_current_status unless ignore_status
+      reload
+    end
+
+    # Reverts stored changes
+    def reload
+      clear_changes
+      @current_start_mode = nil
+    end
+
+    # Determines whether the system has been changed or not
+    #
+    # @return [Boolean] true if the system has been changed
+    def changed?
+      !changes.empty?
+    end
+
+    # Returns the list of supported start modes
+    #
+    # * :on_boot:   The service will be started when the system boots.
+    # * :manual: The service is disabled and it will be started manually.
+    # * :on_demand: The service will be started on demand (using a Systemd socket).
+    #
+    # @return [Array<Symbol>] List of supported modes.
+    def start_modes
+      return @start_modes if @start_modes
+      @start_modes = [:on_boot, :manual]
+      @start_modes << :on_demand if socket
+      @start_modes
+    end
+
+    # Terms to search for this service
+    #
+    # In case the service has an associated socket, the socket name
+    # is included as search term.
+    #
+    # @return [Array<String>] e.g., #=> ["tftp.service", "tftp.socket"]
+    def search_terms
```

Comment:
I do not like API name. It is too much reveal that it is used in journal. Why not call it e.g. "systemd_units"?